### PR TITLE
Continue driving MI Bridges

### PR DESCRIPTION
**Pull request by jayroh**
* Lock Selenium to the current version and monkey-patch   Selenium::WebDriver::Remote::Bridge#execute so that we may slow it   down a little bit (with `sleep`s) for demo purposes. 
* Remove explicit sleeps from within the page objects 
* Add some new steps - mostly intro and starter steps. 
* Create a `click_id` method that will execute javascript to find an   exact html element by id and send a `click()` to it. This helps with   some ambiguous checkbox and radio button labeling/naming 
* Common patterns can be put in the base page object - like `Page#next`
* Moving things into lib, namespaced under `MiBridges::Driver` class which
    does all of the work. This includes setting up capybara and tearing
    things down at the end of the driving session.

### CODE REVIEW

File: lib/mi_bridges/driver.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    private
+
+    attr_reader :snap_application
+
+    def setup
+      Capybara.default_driver = ENV.fetch("WEB_DRIVER", "chrome").to_sym
+    end
+
+    def teardown
+      MiBridges::Driver::BasePage.new.close
```
Comment (jessieay) [inline lib/mi_bridges/driver.rb]: did this fix the session issue I was seeing?

File: lib/mi_bridges/driver.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+      privacy_pin_page = PrivacyPinPage.new
+      privacy_pin_page.fill_in_required_fields
+      privacy_pin_page.next
+
+      BenefitsOverviewPage.new.next
+
+      FraudPenaltyAffidavit.new.next
+
+      start_page = Start.new
```
Comment (jessieay) [inline lib/mi_bridges/driver.rb]: why `Start` rather than `StartPage` ?

File: lib/mi_bridges/driver.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      start_page.fill_in_required_fields
+      start_page.next
+
+      benefits_selector = BenefitsSelector.new
+      benefits_selector.fill_in_required_fields
+      benefits_selector.next
+      benefits_selector.next
```
Comment (jessieay) [inline lib/mi_bridges/driver.rb]: is this second `next` happening on the benefits selector page or is it a separate page? 

if separate, what do you think of creating a page class that represents that view? not worth it, perhaps. But also perhaps confusing to just call it on `benefits_selector` 🤔

File: lib/mi_bridges/driver/base_page.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module MiBridges
+  class Driver
+    class BasePage
```
Comment (jessieay) [inline lib/mi_bridges/driver/base_page.rb]: much better!

File: lib/mi_bridges/driver/benefits_selector.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module MiBridges
+  class Driver
+    class BenefitsSelector < BasePage
+      def fill_in_required_fields
+        click_id("#requestFoodShare")
+        click_id("#isAuthorized_N")
```
Comment (jessieay) [inline lib/mi_bridges/driver/benefits_selector.rb]: what does this id represent? worth it to wrap it into a private method for explicitness?

File: lib/selenium/web_driver/remote.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+              path[key.inspect] = escaper.escape(value.to_s)
+            end
+          rescue IndexError
+            raise ArgumentError,
+              "#{opts.inspect} invalid for #{command.inspect}"
+          end
+
+          WebDriver.logger.info("-> #{verb.to_s.upcase} #{path}")
+
+          case ENV.fetch("DRIVER_SPEED", "fast").to_sym
```
Comment (jessieay) [inline lib/selenium/web_driver/remote.rb]: should we add something in the README about this? Have you tried it yet?

File: lib/selenium/web_driver/remote.rb [hunk 7]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require "selenium-webdriver"
+
+module Selenium
+  module WebDriver
+    module Remote
+      class Bridge
```
Comment (jessieay) [inline lib/selenium/web_driver/remote.rb]: should this file be called `bridge.rb` since that is the class name?

**Comment by jayroh** [comment]
You're the best, Jessie. Thanks for the comments. Will get these whipped into shape this morning :)

File: lib/mi_bridges/driver.rb (hunk 3 above)
Comment (jayroh) [inline lib/mi_bridges/driver.rb]: You're right - it's a completely new page and should be represented with its own object. For sure.

File: lib/mi_bridges/driver.rb (hunk 1 above)
Comment (jayroh) [inline lib/mi_bridges/driver.rb]: I do believe so! Everything looked good for me throughout the day. I'm going to do some work this morning to mitigate that issue, by the way.

File: lib/mi_bridges/driver/benefits_selector.rb (hunk 5 above)
Comment (jayroh) [inline lib/mi_bridges/driver/benefits_selector.rb]: Yeah, once I drive things again this morning I'll do that :)

File: lib/selenium/web_driver/remote.rb (hunk 6 above)
Comment (jayroh) [inline lib/selenium/web_driver/remote.rb]: Yeah, I'll add this to the README now. Also, I did try it and it's A+ awesome. Slows things down *just* the right amount.

**Comment by jayroh** [comment]
[User Agent for Web Driver](https://trello.com/c/WTr8JJL6/1-user-agent-for-web-driver)

**Pull request closed without merging**